front/components/vaults/VaultCreateAppModal.tsx

```diff
@@ -0,0 +1,174 @@
+import {
+  ExclamationCircleStrokeIcon,
+  Input,
+  Modal,
+  Page,
+  RadioButton,
+  TextArea,
+} from "@dust-tt/sparkle";
+import type { APIError, AppVisibility, WorkspaceType } from "@dust-tt/types";
+import { useContext, useState } from "react";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { useApps } from "@app/lib/swr";
+import { MODELS_STRING_MAX_LENGTH } from "@app/lib/utils";
+
+export const VaultCreateAppModal = ({
+  owner,
+  isOpen,
+  setIsOpen,
+}: {
+  owner: WorkspaceType;
+  isOpen: boolean;
+  setIsOpen: (isOpen: boolean) => void;
+}) => {
+  const sendNotification = useContext(SendNotificationsContext);
+
+  const [name, setName] = useState<string | null>(null);
+  const [nameError, setNameError] = useState<string | null>(null);
+
+  const [description, setDescription] = useState<string | null>(null);
+  const [descriptionError, setDescriptionError] = useState<string | null>(null);
+
+  const [visibility, setVisibility] = useState<AppVisibility>("private");
```

Comment:
Do you mean something like? If yes then I think it's better as is 🙈 

```
const [appData, setAppData] = useState({
  name: null,
  nameError: null,
  description: null, 
  descriptionError: null,
  visibility: null
});
```

Reply:
Yes exactly. The goal is to have only one global state that represents the app.

Reply:
I think it makes the state more complex to use and I would rather keep the code as is.

Reply:
I'm going to keep as is because I prefer this way (and this is how we've build all our forms 🙏🏻 ), but if one of you strongly disagree please let me know and we'll discussed it and I'll reopen a PR if needed.

Reply:
I'd say let's at least do like @tdraier suggested 🙃.

Reply:
I don't think it's better because it won't reduce the number of setState. I will open a PR to instead group the error state. Will do for this form and the one to create a folder in a vault.

Reply:
We just need a lib for form validation. 😬